boringtun/src/device/mod.rs

```diff
@@ -637,6 +776,11 @@ impl<T: Tun, S: Sock> Device<T, S> {
                         Some(peer) => peer,
                     };
 
+                    let mut public_key = String::with_capacity(32);
+                    for byte in peer.tunnel.peer_static_public().as_bytes() {
+                        let pub_symbol = format!("{:02X}", byte);
+                        public_key.push_str(&pub_symbol);
```

Comment:
you can use `write!(public_key, "{:02X}", byte);` this will avoid uneeded alloc

Reply:
or event better `write!("{:X}", peer.tunnel.peer_static_public().as_bytes())`;

Reply:
The latter does not work because `[u8; 32]` does not implement `LowerHex` trait

Reply:
Done